stdlib/public/core/StringBridge.swift

```diff
@@ -337,42 +360,59 @@ internal func _cocoaASCIIPointer(_ str: _CocoaString) -> UnsafePointer<UInt8>? {
   return _NSStringASCIIPointer(_objc(str))
 }
 
-private enum CocoaStringPointer {
-  case ascii(UnsafePointer<UInt8>)
-  case utf8(UnsafePointer<UInt8>)
-  case utf16(UnsafePointer<UInt16>)
-  case none
-}
-
-@_effects(readonly)
-private func _getCocoaStringPointer(
-  _ cfImmutableValue: _CocoaString
-) -> CocoaStringPointer {
-  if let asciiPtr = _cocoaASCIIPointer(cfImmutableValue) {
-    // NOTE: CFStringGetCStringPointer means ASCII
-    return .ascii(asciiPtr)
+// This does not create a bridged String that's safe to use past the current
+// stack frame, e.g. it does not defensively copy it and so on.
+@_effects(releasenone)
+internal func _withTemporaryBridgedCocoaString<R>(
+  _ cocoaString: _CocoaString?,
+  _ work: (String) throws -> R
+) rethrows -> R {
+  guard let cocoaString = cocoaString else {
+    return try work("")
```

Comment:
Why not have it be non-optional and the callers can guard-else-return? In general, we like to shed optionality as early as reasonable so that core logic operates over existing values. This leaves nil-handling to the interface layer.